cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/CgmesBoundary.java

```diff
@@ -110,13 +131,65 @@ public String nameAtBoundary(String node) {
         return nodesName.containsKey(node) ? nodesName.get(node) : "XnodeCode-unknown";
     }
 
+    public static final class BoundaryEquipment {
+        private BoundaryEquipment(BoundaryEquipmentType type, PropertyBag propertyBag) {
+            this.type = type;
+            this.propertyBag = propertyBag;
+            propertyBags = null;
+        }
+
+        private BoundaryEquipment(BoundaryEquipmentType type, PropertyBags propertyBags) {
+            this.type = type;
+            this.propertyBag = null;
+            this.propertyBags = propertyBags;
+        }
+
+        public BoundaryEquipmentType getBoundaryEquipmentType() {
+            return type;
+        }
+
+        public EquipmentAtBoundaryConversion createConversion(Context context) {
+            EquipmentAtBoundaryConversion c = null;
+            switch (type) {
+                case AC_LINE_SEGMENT:
+                    c = new ACLineSegmentConversion(propertyBag, context);
+                    break;
+                case SWITCH:
+                    c = new SwitchConversion(propertyBag, context);
+                    break;
+                case TRANSFORMER:
+                    c = new TwoWindingsTransformerConversion(propertyBags, context);
+                    break;
+                case EQUIVALENT_BRANCH:
+                    c = new EquivalentBranchConversion(propertyBag, context);
+                    break;
+            }
+            return c;
+        }
+
+        public void log() {
+            if (LOG.isDebugEnabled()) {
+                if (propertyBag != null) {
+                    LOG.debug(propertyBag.tabulateLocals(type.toString()));
+                }
+                if (propertyBags != null) {
+                    propertyBags.forEach(p -> LOG.debug(p.tabulateLocals(type.toString())));
+                }
+            }
+        }
+
+        private final BoundaryEquipmentType type;
+        private final PropertyBag propertyBag;
+        private final PropertyBags propertyBags;
+    }
```

Comment:
I think put this code as an external class

Reply:
Done